### PR TITLE
USSD Public functionality - Opt out

### PR DESCRIPTION
**Pull request by gsvr**

### CODE REVIEW

**Comment by gsvr** [comment]
@imsickofmaps Please review.

File: src/ussd_public.js [hunk 1]
```diff
@@ -581,6 +619,66 @@ go.app = function() {
             });
         });
 
+        self.add('state_optout_all', function(name) {
+            if (self.im.user.answers.household_id === null) {
```
Comment (imsickofmaps) [inline src/ussd_public.js:641]: is this definitely `null` not `undefined`? or can it be either?

File: src/utils.js [hunk 2]
```diff
@@ -487,6 +487,21 @@ go.utils = {
...
+    optout: function(im, identity_id, optout_reason) {
+      // Posts an optout with a reason to the identity store optout endpoint
+
+        // TODO: use correct optout data
+        var optout_info = {
+            identity: identity_id,
+            reason: optout_reason || 'unknown'
+        };
+        return go.utils
+            .service_api_call("optout", "post", null, optout_info, "optout/", im)
```
Comment (imsickofmaps) [inline src/utils.js]: The `service` is still identity here

File: src/utils_project.js [hunk 3]
```diff
@@ -511,30 +511,73 @@ go.utils_project = {
...
 
-            return Q.all([
-                // update mama identity
-                go.utils.update_identity(im, mama_identity),
-                // update subscription
-                go.utils_project.update_subscription(im, subscription)
-            ]);
-        });
+        return go.utils
+            .service_api_call("change", "post", null, change_data, "change/", im)
```
Comment (imsickofmaps) [inline src/utils_project.js]: the `service` is still registration here and else where this endpoint is used

**Comment by imsickofmaps** [comment]
@gsvr I was like "where's all the tests!" then there was a moment of swelling TDD pride: 
screenshot 2016-03-17 13 32 08

File: src/ussd_public.js (hunk 1 above)
Comment (gsvr) [inline src/ussd_public.js:641]: I explicitly set it to `null` in `state_check_receiver_role`. If it is `undefined` something would have gone wrong and I don't want to try acting on that.

**Comment by gsvr** [comment]
@imsickofmaps Second pass please.

**Comment by imsickofmaps** [comment]
@gsvr :+1:  good work! This turned into a monster task though, let's try to remember to talk about "right sizing" this kind of thing into chunks in retro on Tuesday.

**Pull request merged**